src/main/java/stirling/software/SPDF/controller/web/AccountWebController.java

```diff
@@ -59,132 +71,129 @@ public String login(HttpServletRequest request, Model model, Authentication auth
+
         if (oauth != null) {
             if (oauth.getEnabled()) {
                 if (oauth.isSettingsValid()) {
-                    providerList.put("/oauth2/authorization/oidc", oauth.getProvider());
+                    String firstChar = String.valueOf(oauth.getProvider().charAt(0));
+                    String clientName =
+                            oauth.getProvider().replaceFirst(firstChar, firstChar.toUpperCase());
+                    providerList.put(OAUTH_2_AUTHORIZATION + oauth.getProvider(), clientName);
                 }
+
                 Client client = oauth.getClient();
+
                 if (client != null) {
                     GoogleProvider google = client.getGoogle();
-                    if (google.isSettingsValid()) {
+
+                    if (validateProvider(google)) {
                         providerList.put(
-                                "/oauth2/authorization/" + google.getName(),
-                                google.getClientName());
+                                OAUTH_2_AUTHORIZATION + google.getName(), google.getClientName());
                     }
-                    GithubProvider github = client.getGithub();
-                    if (github.isSettingsValid()) {
+
+                    GitHubProvider github = client.getGithub();
+
+                    if (validateProvider(github)) {
                         providerList.put(
-                                "/oauth2/authorization/" + github.getName(),
-                                github.getClientName());
+                                OAUTH_2_AUTHORIZATION + github.getName(), github.getClientName());
                     }
+
                     KeycloakProvider keycloak = client.getKeycloak();
-                    if (keycloak.isSettingsValid()) {
+
+                    if (validateProvider(keycloak)) {
                         providerList.put(
-                                "/oauth2/authorization/" + keycloak.getName(),
+                                OAUTH_2_AUTHORIZATION + keycloak.getName(),
                                 keycloak.getClientName());
                     }
                 }
             }
         }
+
         SAML2 saml2 = securityProps.getSaml2();
-        if (securityProps.isSaml2Activ()
-                && applicationProperties.getSystem().getEnableAlphaFunctionality()) {
-            providerList.put("/saml2/authenticate/" + saml2.getRegistrationId(), "SAML 2");
+
+        if (securityProps.isSaml2Active()
+                && applicationProperties.getSystem().getEnableAlphaFunctionality()
+                && applicationProperties.getEnterpriseEdition().isEnabled()) {
+            String samlIdp = saml2.getProvider();
+            String saml2AuthenticationPath = "/saml2/authenticate/" + saml2.getRegistrationId();
+
+            if (applicationProperties.getEnterpriseEdition().isSsoAutoLogin()) {
+                return "redirect:"
+                        + SPDFApplication.getStaticBaseUrl()
```

Comment:
I dont believe this works with reverse proxies and similar which will use a different URL and port vs what the host uses.
I believe this redirect should be context-aware to the address used in the session

Reply:
Good point, I changed it to get the address from the request now